README.md

```diff
@@ -1,5 +1,38 @@
-# lighthouse 
-Controller to facilitate DNS discovery in multi-cluster connected environments (proof-of-concept state)
+# Lighthouse 
+The lighthouse along with Submariner facilitate DNS discovery in multi-cluster connected environments. The solution aims to compatible with any CNI-compatible cluster network provider.
+
+# Architecture
+The below digram shows the basic lighthouse architecture.
+
+![Lighthouse architecture](https://raw.githubusercontent.com/submariner-io/lighthouse/master/docs/img/architecture.png) 
+
+## Lighthouse Controller
+There is a central service discovery controller that gathers the information from the clusters, decides what information is to be shared and distributes the information as newly defined CRDs.
+
+The lighthouse controller will be running along with the admiral control plane. It uses the API’s exposed by the admiral to connect to each cluster and listen for service CRUD. Once it is notified that a new service is created and if the service needs to be distributed, it will frame the lighthouse CRD and the same will be distributed to all the clusters that are part of the control plane. The flow for update and delete will be similar.
+
+### WorkFlow
+A typical work flow can be as follows.
+
+- Lighthouse controller registers with Admiral for cluster join and unjoin events.
+- Once it is notified about a join event, it retrieves the credentials from Admiral and registers for call back on service creation with a shared informer.
+- When a new service is created in a cluster, lighthouse controller will be notified about it and it will create the Lighthouse CRD with the service info.
+- The controller with the help of Admiral distributes the CRD to all the clusters that have joined.
+
+![Lighthouse Controller WorkFlow](https://raw.githubusercontent.com/submariner-io/lighthouse/master/docs/img/controllerWorkFlow.png) 
+
+## Lighthouse Plugin
+The Lighthouse CoreDNS plugin will work along with the default kubernetes plugin. It will use the Lighthouse CRD that is distributed by the controller for DNS resolution. The below diagram indicates a high level architecture.
```

Comment:
add workflow doc after this line.

Reply:
There was another diagram supposed to be here and missed it. Now added the same.